engine/src/engine.rs

```diff
@@ -490,13 +483,7 @@ impl<'env, I: IO + Copy, E: Env> Engine<'env, I, E> {
         };
 
         let used_gas = executor.used_gas();
-        let status = match exit_reason.into_result(result) {
-            Ok(status) => status,
-            Err(e) => {
```

Comment:
Is it correct that we don't increment the nonce in case of any error?

Reply:
I'm double checking.

Reply:
Removing this code is fine because we were not actually incrementing the nonce in this case even though it looks like we were. In `lib.rs` we are unwrapping the `Result` from this function (and similarly for `call` and `submit`), so the contract will panic when we return an `Err` variant, thus reverting any state changes we made.